core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -44,9 +44,9 @@
 
     public static final String NAME = "boosting";
 
-    private QueryBuilder positiveQuery;
+    private QueryBuilder positiveQuery = EmptyQueryBuilder.PROTOTYPE;
 
-    private QueryBuilder negativeQuery;
+    private QueryBuilder negativeQuery = EmptyQueryBuilder.PROTOTYPE;
```

Comment:
I find a bit confusing how we handle the boosting query: the parser validates that the json contains the "negative" and "positive" elements. That said they can be empty, which would map to the new empty query builder. For java api, default is the empty query builder and one is not required to set negative and positive, no validation happens. I think that we should either require setting those two queries in both parser and builder, or remove any validation around them. 

For instance, I would go and remove the default value here, and remove validation from the parser around whether negative and positive are part of the query. Then add a check in validate method that positive and negative need to be non null. This way the behaviour stays the same as before when it comes to json (no element means no set, which means validation error), but also java api users have to remember to set the queries. Thoughts?